### PR TITLE
Add clang-17 to ci

### LINKED ISSUE #1360: [FEA]: llvm / clang 17 CI coverage
### Is this a duplicate?

- [ ] I confirmed there appear to be no [duplicate issues](https://github.com/NVIDIA/cccl/issues) for this request and that I agree to the [Code of Conduct](CODE_OF_CONDUCT.md)

### Area

libcu++

### Is your feature request related to a problem? Please describe.

Hello! We see that in CI you are covering the binaries from the LLVM 16.x releases, but there is no coverage yet for the 17.x releases:

https://github.com/NVIDIA/cccl/blob/c494fe94fbc5b83c517613ad671656a35340f829/ci/matrix.yaml#L34

That's something that is very interesting for us.

Anecdote: we are debugging a clang-tidy error and this actually related to clang being tripped up by code in `libcudacxx/cuda/std/detail/libcxx/include` and for our debugging it was interesting to see where you are in terms of supporting clang 17.

### Describe the solution you'd like

cover clang 17 in your CI

### Describe alternatives you've considered

n/a

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by miscco**
Fixes #1360

### CODE REVIEW

**Comment by jgehrcke** [comment]
Much appreciated! :heart:

File: ci/matrix.yaml [hunk 1]
```diff
@@ -30,8 +30,9 @@ llvm13: &llvm13 { name: 'llvm', version: '13', exe: 'clang++' }
 llvm14: &llvm14 { name: 'llvm', version: '14', exe: 'clang++' }
 llvm15: &llvm15 { name: 'llvm', version: '15', exe: 'clang++' }
 llvm16: &llvm16 { name: 'llvm', version: '16', exe: 'clang++' }
+llvm16: &llvm17 { name: 'llvm', version: '17', exe: 'clang++' }
```
Comment (jrhemstad) [inline ci/matrix.yaml:33]:
```suggestion
llvm17: &llvm17 { name: 'llvm', version: '17', exe: 'clang++' }
```

**Comment by jrhemstad** [comment]
Also, this won't work because clang17 isn't a supported host compiler for CUDA 12.3.

**Comment by miscco** [comment]
Already done

**Pull request closed without merging**